features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/BitmapPool.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.base64
+
+import android.content.ComponentCallbacks2
+import android.content.res.Configuration
+import android.graphics.Bitmap
+import android.graphics.Bitmap.Config
+import android.util.LruCache
+import com.datadog.android.sessionreplay.internal.utils.CacheUtils
+
+internal object BitmapPool : Cache<String, Bitmap>, ComponentCallbacks2 {
+    @Suppress("MagicNumber")
+    val MAX_CACHE_MEMORY_SIZE_BYTES = (Runtime.getRuntime().maxMemory() / 8).toInt()
+
+    private var cache: LruCache<String, Bitmap> = object :
+        LruCache<String, Bitmap>(MAX_CACHE_MEMORY_SIZE_BYTES) {
+        override fun sizeOf(key: String?, value: Bitmap): Int {
+            return value.allocationByteCount
+        }
+    }
+
+    override fun put(value: Bitmap) {
+        val key = generateKey(value)
+        cache.put(key, value)
+    }
+
+    override fun size(): Int = cache.size()
+
+    override fun clear() = cache.evictAll()
+
+    override fun get(element: String): Bitmap? = cache.get(element)
```

Comment:
the way I see this `BitmapPool` will be shared between threads. How will you make sure that you do not have race issues (a bitmap needs only to be re - used by one single thread at a time) ?